homeassistant/components/system_bridge/coordinator.py

```diff
@@ -49,97 +73,110 @@ def update_listeners(self) -> None:
         for update_callback in self._listeners:
             update_callback()
 
-    async def async_handle_event(self, event: Event):
-        """Handle System Bridge events from the WebSocket."""
-        # No need to update anything, as everything is updated in the caller
-        self.logger.debug(
-            "New event from %s (%s): %s", self.title, self.host, event.name
-        )
-        self.async_set_updated_data(self.bridge)
-
-    async def _listen_for_events(self) -> None:
+    async def async_handle_data(
+        self,
+        module: str,
+        data: dict,
+    ) -> None:
+        """Handle data from the WebSocket client."""
+        if module in MODULES:
+            self.logger.debug("Set new data for: %s", module)
+            if module_handler := MODEL_MAP.get(module):
+                self.logger.debug("Module handler : %s", module_handler)
+                setattr(self.systembridge_data, module, module_handler(**data))
```

Comment:
Why doesn't the library call `async_handle_data` with the appropriate instances already? The module map and handlers are details of the library in my opinion.